shaccson/src/main/java/UserDao.java

```diff
@@ -0,0 +1,121 @@
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+
+    private static final String CREATE_USER_QUERY =
+            "INSERT INTO users(username, email, password) VALUES (?, ?, ?)";
+    private static final String READ_USER_QUERY =
+            "SELECT * FROM users WHERE id = ?";
+    private static final String UPDATE_USER_QUERY =
+            "UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?";
+    private static final String DELETE_USER_QUERY =
+            "DELETE FROM users WHERE id = ?";
+    private static final String SELECT_ALL_USERS_QUERY =
+            "SELECT * FROM users";
+
+    public User addUser(final User user) {
+        try (final Connection conn = DbUtil.getConnection();
+             final PreparedStatement statement = conn.prepareStatement(CREATE_USER_QUERY, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getUsername());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.executeUpdate();
+
+            try (final ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    user.setId(generatedKeys.getInt(1));
+                }
+            }
+        } catch (final SQLException e) {
+            System.err.println("Error while adding user: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+        return null;
+    }
+
+    public User getUserById(final int id) {
+        try (final Connection conn = DbUtil.getConnection();
+             final PreparedStatement statement = conn.prepareStatement(READ_USER_QUERY)) {
+
+            statement.setInt(1, id);
+            try (final ResultSet rs = statement.executeQuery()) {
+                if (rs.next()) {
+                    final User user = new User();
+                    user.setId(rs.getInt("id"));
+                    user.setUsername(rs.getString("username"));
+                    user.setEmail(rs.getString("email"));
+                    user.setPassword(rs.getString("password"));
+                    return user;
+                }
+            }
+        } catch (final SQLException e) {
+            System.err.println("Error while retrieving user by ID: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+        return null;
+    }
+
+    public void updateUser(final User user) {
+        try (final Connection conn = DbUtil.getConnection();
+             final PreparedStatement statement = conn.prepareStatement(UPDATE_USER_QUERY)) {
+            statement.setString(1, user.getUsername());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
```

Comment:
BUG: zaheshovat jen nova hesla, pri aktualizaci stareho nechceme hashovat znova zaheshovane heslo.

```suggestion
statement.setString(3, user.getPassword());
```

Reply:
opraveno